### PR TITLE
FocalPointPicker: Fix layout misalignment when placed in the `BlockInspector`

### PR DESCRIPTION
**Pull request by t-hamano**
## What?
This PR fixes a problem with the unit controls (Left, Top) not being aligned vertically when the `FocalPointPicker` component is used in the block sidebar (`InspectorControls`).

| Before | After |
| --- | --- |
| before | after |

## Why?

The control itself has no margin, but `margin-bottom` is applied by the `InspectorControl`. And this is because the value is smaller in the UnitControl on the right side.

focalpointpicker

## How?

I have reset the `margin-bottom` on the `UnitControl`. I believe this is the appropriate approach as this margin is not expected, but if there is another way, please advise.

## Testing Instructions

- Open the post/site editor.
- Insert a cover or media & text block, which has this component in the sidebar.
- Confirm that the vertical positions of the left and top controls are aligned.

### CODE REVIEW

File: packages/components/CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,7 @@
 -   `InputControl`: Fix internal `Flex` wrapper usage that could add an unintended `height: 100%` ([#46213](https://github.com/WordPress/gutenberg/pull/46213)).
 -   `Navigator`: Allow calling `goTo` and `goBack` twice in one render cycle ([#46391](https://github.com/WordPress/gutenberg/pull/46391)).
 -   `Modal`: Fix unexpected modal closing in IME Composition ([#46453](https://github.com/WordPress/gutenberg/pull/46453)).
+-   `FocalPointPicker`: Reset `margin-bottom` of `UnitControl` to align vertical position ([#46631](https://github.com/WordPress/gutenberg/pull/46631)).
```
Comment (t-hamano) [inline packages/components/CHANGELOG.md]: Is this PR a bug fix? Maybe it is 'internal' or 'enhancement'...

**Comment by glendaviesnz** [comment]
Should we add the `[__nextHasNoMarginBottom](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/focal-point-picker/index.tsx#L86)` prop to UnitControl to control this? This is used by the parent FocalPicker and a number of other blocks to remove the bottom margin. @mirka, is there a reason why the UnitControl doesn't currently use this prop, or is it just a case of it not making it to the top of the to-do list yet?

**Comment by mirka** [comment]
Mmm, good catch. This one is caused by the problematic [override style](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/block-inspector/style.scss#L13-L19) added in the block-editor package's `BlockInspector`. We want to eventually remove these override styles once we have dedicated layout components (#43423), but the migration will take a long time.

So yeah @glendaviesnz, the FocalPointPicker itself is already being used with the no-margin prop enabled, but the consumer (`block-editor` in this case) is adding interfering styles.

@t-hamano If possible, my preference would be to add any workarounds in the offending consumer package (`block-editor`), rather than in the `components` package. So like, scope the workaround styles to `.block-editor-block-inspector .components-focal-point-picker-control .components-base-control` or something, and put them right below the [styles](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/block-inspector/style.scss#L13-L19) we're trying to counteract. This way the messiness will be colocated in one place, and it's clearer that those hacks are related.

File: packages/components/CHANGELOG.md (hunk 1 above)
Comment (mirka) [reply packages/components/CHANGELOG.md]: I would classify it as a bug fix.

Comment (t-hamano) [reply packages/components/CHANGELOG.md]: I have moved the changelog entry to the `block-editor` package.

**Comment by t-hamano** [comment]
@mirka 
Thanks for the advice!
Indeed, I think that approach makes sense. I have moved the CSS overrides (and the changelog) to the `block-editor` package.

focal-point-picker

**Review by mirka — APPROVED** [review]
Looks good, thanks! 🚢

**Pull request merged**